### PR TITLE
feat: don't authorize account creation for unconfigured apps

### PR DESCRIPTION
**Pull request by beeman**

### CODE REVIEW

**Comment by cat-turner** [comment]
unit test TestAuthorizer_RateLimiter seems to fail. Can you please resolve this test failure?

File: pkg/account/authorizer.go [hunk 1]
```diff
@@ -226,14 +228,19 @@ func (a *authorizer) Authorize(ctx context.Context, tx solana.Transaction) (resu
 
 	cfg, err := a.config.Get(ctx, appIndex)
 	if err == app.ErrNotFound || cfg == nil || cfg.CreateAccountURL == nil {
+		if err == app.ErrNotFound {
```
Comment (cat-turner) [inline pkg/account/authorizer.go:231]: I recall you mentioned that if there is no app URL (where the wallet was created) we should throw an error. Does that mean you should only check `cfg.CreateAccountURL`? Why are you checking cfg?

**Review by cat-turner — COMMENTED** [review]
Looks good but I think it would be good to address failing unit test before we merge. I re-ran the test on CI and it failed, so this doesn't seem like a flakey test. If you want to keep all conditions you include in line 230, then the unit test may need to be updated.

**Comment by beeman** [comment]
Tests are now fixed by making sure the apps used in the tests have a config.

https://github.com/kinecosystem/agora/pull/12/files#diff-aebe417e718c8358f06d53ce3d05fc2791d521d1f3b90d74841a722c8b1d8adbR311-R312

File: pkg/account/authorizer.go (hunk 1 above)
Comment (beeman) [reply pkg/account/authorizer.go:231]: The `cfg.CreateAccountURL` is not in play here, we are throwing an error for apps that are un-configured.

**Pull request merged**